governance-with-multisig/README.md

```diff
@@ -0,0 +1,136 @@
+# Upgradeability governance via multisig
+
+This project experiments with managing the upgradeability of a project via a multisig. Instead of a single Ethereum account having full control over when a contract should be upgraded, this responsibility is decentralized to multiple parties. In this scenario, these parties coordinate their actions via a multi-signature wallet. This allows for setups where a predefined group of accounts have to agree upon an upgrade before it can be actually executed.
+
+This approach requires **no changes** at all on ZeppelinOS, and uses [the Gnosis MultiSig wallet implementation](https://github.com/gnosis/MultiSigWallet) for governance, already [audited by the Zeppelin team](https://blog.zeppelin.solutions/gnosis-multisig-wallet-audit-d702ff0e2b1e). The only caveat is that upgradeability must be managed manually by the MultiSig owners, altough all other operations (including the deployment of new logic contracts) are still manged via the ZeppelinOS CLI.
+
+## About this project
+
+You'll find two test contracts in this project: `EthBox` and `EthBoxV2`. The first has a single `deposit` method that stores ETH sent from an account. The latter extends the former, by adding a `getBalance` function to query the balance of an account. The goal will be to deploy an instance of `EthBox`, transfer upgradeability rights to a multisig wallet, and require the consensus of the multisig owners to upgrade it to `EthBoxV2`.
+
+## About this guide
+
+This guide will take you through the following steps:
+
+1- Set up a new ZeppelinOS project
+2- Register a contract and deploy it to a network
+3- Create a new instance of your contract
+4- Create a multisig wallet and transfer upgradeability rights of the new contract instance to it
+5- Register a new version of your logic contract and deploy it
+6- Submit a request to the multisig wallet to upgrade the contract instance
+7- Confirm the request and check the contract was upgraded
+
+## Step by step
+
+We'll begin by setting up the ZeppelinOS project. Clone this repository, install dependencies, and initialize the `zos` project to get started.
+
+```bash
+$ git clone https://github.com/zeppelinos/labs.git
+$ cd labs/governance-with-multisig
+$ npm install
+$ npx zos init governance-with-multisig
+```
+
+Our first step will be to register the `EthBox` contract in our project, deploy the logic contract to the blockchain, and create a new instance. We'll be working on a local ganache instance (you can start one via `./ganache-sh`), though you can also run this on any real network.
+
+```sh
+$ npx zos add EthBox
+$ npx zos push --skip-compile --network local
+$ npx zos create EthBox --network local
+```
+
+We can test that the EthBox instance was correctly deployed by `deposit`ing some ETH on it from a `truffle console`. Replace `MY_ETHBOX_ADDRESS` with the address returned by the last command run above.
```

Comment:
`EthBox`

Reply:
Or use `EthBox.deployed()` once new minor version of `zos-cli` is released ;)

Reply:
Yep, but since I couldn't use `EthBoxV2.deployed()` later, I didn't want to introduce this